cmp/path.go

```diff
@@ -188,17 +188,6 @@ type (
 		typ reflect.Type
 	}
```

Comment:
Curious, why not have the methods right below each of these type definitions as well?